python/server/ServerWrapper.cpp

```diff
@@ -475,18 +524,8 @@ namespace {
         int SpawnLimit()
         { return m_monster_fleet_plan->SpawnLimit(); }
 
-        bool Location(int object_id) {
-            // get the universe object to test and check if it exists
-            TemporaryPtr<UniverseObject> obj = GetUniverseObject(object_id);
-            if (!obj) {
-                ErrorLogger() << "MonsterFleetPlanWrapper::Location: Couldn't get object with ID " << object_id;
-                return false;
-            }
-
-            // get location condition and evaluate it with the specified universe object
-            // if no location condition has been defined, any object matches
-            const Condition::ConditionBase* location_test = m_monster_fleet_plan->Location();
-            return (!location_test || location_test->Eval(obj));
+        list Locations(list systems) {
+            return FilterIDsWithCondition(m_monster_fleet_plan->Location(), systems);
```

Comment:
`Locations()` is orders of magnitude faster than `Location()` .  Checking all of the systems at once with `Locations()` is just as fast as checking one location with `Location()`.

Removing `Location()` encourages vectorizing those tests, which is much faster.

This is high-handed, but I think that the difference is large enough to justify it.

For example replacing https://github.com/freeorion/freeorion/pull/1225/files/d8707e3f267a172e7293311847c92d54c427c858#diff-f6a824c38bd1953f33a55ae2606fc199R146,

`fp_location_cache[fleet_plan] = set(fleet_plan.locations(systems))`

with a loop adds 20.5s to the startup time of a 999 galaxy.